kong/plugins/hmac-auth/access.lua

```diff
@@ -95,82 +94,87 @@ local function validate_params(params, conf)
   return nil, fmt("algorithm %s not supported", params.algorithm)
 end
 
-local function retrieve_hmac_fields(request, headers, header_name, conf)
+
+local function retrieve_hmac_fields(authorization_header)
   local hmac_params = {}
-  local authorization_header = headers[header_name]
+
   -- parse the header to retrieve hamc parameters
   if authorization_header then
-    local iterator, iter_err = ngx_gmatch(authorization_header, "\\s*[Hh]mac\\s*username=\"(.+)\",\\s*algorithm=\"(.+)\",\\s*headers=\"(.+)\",\\s*signature=\"(.+)\"")
+    local iterator, iter_err = re_gmatch(authorization_header,
+                                          "\\s*[Hh]mac\\s*username=\"(.+)\",\\s*algorithm=\"(.+)\",\\s*headers=\"(.+)\",\\s*signature=\"(.+)\"")
```

Comment:
you can reduce the line by breaking this long string into shorter ones connected by `..` (don't worry, the compiler's constant-subexpression-elimination will make it into a single string)